File: insertionSort.js
const insertionSort = arr => {
  for(let i = 0; i < arr.length; i++) {
    let temp = arr[i];
    let j = i - 1;
    while (j >= 0 && arr[j] > temp) {
      arr[j+1] = arr[j];
      j--;
    }
    arr[j+1] = temp;
  }
  return arr;
};

const array = [2,5,6,9,4,3,7,10,1,8];

console.log(insertionSort(array));

File: insertion_sort.rb
def insert_sort(arr)
  arr.each_with_index do |el, idx|
    temp = el
    j = idx - 1
    while arr[j] > temp && j >= 0
      arr[j+1] = arr[j]
      j -= 1
    end
    arr[j+1] = temp
  end
end

array = [1,2,9,4,5]
p insert_sort(array)
